lib/gen_stage.ex

```diff
@@ -744,19 +744,20 @@ defmodule GenStage do
   @type type :: :producer | :consumer | :producer_consumer
 
   @typedoc "Options used by the `subscribe*` functions"
-  @type subscription_options :: keyword
-    
-  @typedoc "Option values used by the `init*` functions when stage type is `:producer_consumer`"
-  @type producer_consumer_option :: {:buffer_size, non_neg_integer | :infinity} |
-                           {:buffer_keep, :first | :last} |
-                           {:dispatcher, module | {module, GenStage.Dispatcher.options}}
+  @type subscription_options :: {:cancel, :permanent | :transient | :temporary} |
+                           {:min_demand, integer} |
+                           {:max_demand, integer}
 
   @typedoc "Option values used by the `init*` functions when stage type is `:producer`"
-  @type producer_option :: {:demand, :forward | :accumulate} | producer_consumer_option
-  
-  @typedoc "Option values used by the `init*` functions when stage type is `:consumer`"
-  @type consumer_option :: {:subscribe_to, [module | {module, subscription_options}]} | producer_consumer_option
+  @type producer_option :: {:demand, :forward | :accumulate}
```

Comment:
`producer_option` should have all producer options so another module can use `GenStage.producer_option` to refer to all producer options. It is fine for it to reference other types though.

Reply:
Ok, PR updated